tests/instancetype/instancetype.go

```diff
@@ -238,6 +241,90 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			memRequest := vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory]
 			Expect(memRequest.Value()).To(Equal(expectedOverhead))
 		})
+		It("should apply 80% memory overcommit from instancetype without a preference", func() {
+			vmi := libvmifact.NewAlpine()
+
+			instancetype := builder.NewInstancetypeFromVMI(vmi)
+			instancetype.Spec.Memory.OvercommitPercent = 80
+
+			instancetype, err := virtClient.VirtualMachineInstancetype(testsuite.GetTestNamespace(instancetype)).
+				Create(context.Background(), instancetype, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vm := libvmi.NewVirtualMachine(vmi,
+				libvmi.WithInstancetype(instancetype.Name),
+				libvmi.WithRunStrategy(virtv1.RunStrategyAlways),
+			)
+			vm, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).
+				Create(context.Background(), vm, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(matcher.ThisVM(vm)).WithTimeout(timeout).WithPolling(time.Second).
+				Should(matcher.BeReady())
+
+			vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).
+				Get(context.Background(), vm.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(*vmi.Spec.Domain.Memory.Guest).To(Equal(instancetype.Spec.Memory.Guest))
+
+			expectedRequest := int64(float32(instancetype.Spec.Memory.Guest.Value()) * (1 - float32(instancetype.Spec.Memory.OvercommitPercent)/100))
+			memRequest := vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory]
+			Expect(memRequest.Value()).To(Equal(expectedRequest))
+		})
+		It("should apply 80% memory overcommit from instancetype to VMI with preference requirements", func() {
```

Comment:
What's the difference between this case and `should apply memory overcommit instancetype to VMI even with cluster overcommit set`, can they not be merged?

Reply:
`should apply memory overcommit instancetype to VMI even with cluster overcommit set` -- this test has overcommit value 15 and preference is created with empty spec so it does not influence the memory calculation.
`should apply 80% memory overcommit from instancetype to VMI with preference requirements` --  in this test,  i am creating spec.requirements.cpu.guest and spec.requirements.memory.guest explictly. This test confirms vmi memory is still calculated based on the instancetype.